common/djangoapps/student/views.py

```diff
@@ -1236,9 +1236,22 @@ def login_user(request, error=""):  # pylint: disable=too-many-statements,unused
                 AUDIT_LOG.warning(u"Login failed - password for user.id: {0} is invalid".format(loggable_id))
             else:
                 AUDIT_LOG.warning(u"Login failed - password for {0} is invalid".format(email))
+
+        # Labster changes
+        # If there was login error, suggest student to login to UK region server
+        site_name = configuration_helpers.get_value('SITE_NAME', settings.SITE_NAME)
+        if not site_name == 'courses-uk.labster.com':
```

Comment:
Can we place `courses-uk.labster.com` in lms/env/labster.py?
e.g.
```py
REGIONAL_EDX = 'courses-uk.labster.com'
```

Reply:
As you might know, we might need to create a separate Regioanl servers for the students from other regions (like Canada etc.). 
And this solution will require additional development to support other regions too.
Can we simply use [CountryMiddleware](https://github.com/edx/edx-platform/blob/master/lms/envs/common.py#L1205) that adds [country code to the session](https://github.com/edx/edx-platform/blob/e8a36957b1f732974260e7b9b42b9c25148b492c/openedx/core/djangoapps/geoinfo/middleware.py#L44) ?

Chnages in `lms/envs/labster.py`
```py
REGIONS = {
  'uk': {'url': 'https://courses-uk.labster.com'},
}
```
In this file:
```py
regions = configuration_helpers.get_value('REGIONS', settings.REGIONS)
if request.session['country_code'] in regions:
  region_info = regions[request.session['country_code']]
  msg = _('...').format(region_url=region_info['url'])
else:
  msg = _('Email or password is incorrect.')
```
In this case, DevOps just need to add a new region info to the settings and it will work without additional code changes.
Note: Do not forget to make settings managable for the DevOps.